frontend/src/components/DragAndDrop/DragAndDrop.js

```diff
@@ -0,0 +1,146 @@
+import { useState, useRef, useEffect } from "react";
+import { CloudArrowUpFill } from "react-bootstrap-icons";
+import "./DragAndDrop.css";
+import NavBar from "../NavBar";
+import App from "../../App";
+
+function DragAndDrop() {
+  const [dragActive, setDragActive] = useState(false);
+  const [invalidFile, setInvalidFile] = useState(false);
+  const [lines, setLines] = useState();
+  const inputRef = useRef(null);
+
+  const handleDragEnter = function (e) {
+    e.preventDefault();
+    e.stopPropagation();
+    setDragActive(true);
+  };
+
+  const handleDragLeave = function (e) {
+    e.preventDefault();
+    e.stopPropagation();
+    setDragActive(false);
+  };
+
+  const handleDrop = function (e) {
+    e.preventDefault();
+    e.stopPropagation();
+    setDragActive(false);
+    if (e.dataTransfer.files && e.dataTransfer.files[0]) {
+      const file = e.dataTransfer.files[0];
+      handleFiles(file);
+    }
+  };
+
+  const handleChange = function (e) {
+    e.preventDefault();
+    e.stopPropagation();
+    if (e.target.files && e.target.files[0]) {
+      const file = e.target.files[0];
+      handleFiles(file);
+    }
+  };
+
+  const onButtonClick = () => {
+    inputRef.current.value = "";
+    inputRef.current.click();
+  };
+
+  const handleFiles = (file) => {
+    const reader = new FileReader();
+    reader.onload = (e) => {
+      try {
+        const dataObjectsArray = e.target.result.trim().split(/\r?\n/);
+        const lines = dataObjectsArray.map((line) => JSON.parse(line));
+        setLines(lines);
+      } catch (error) {
+        setInvalidFile(true);
+        setTimeout(() => {
+          setDragActive(false);
+          setInvalidFile(false);
+        }, 4000);
+        console.error("Error :", error);
+      }
+    };
+    reader.readAsText(file);
+  };
+
+  return (
+    <div>
+      {lines ? (
+        <App lines={lines} />
+      ) : (
+        <div>
+          <NavBar />
+          <div
+            aria-live="polite"
+            aria-atomic="true"
+            style={{
```

Comment:
(As usual not being a React or even frontend dev) -- is there a reason you use inline styles here but still have a separate CSS file?

Reply:
Actually, this is not even required anymore! Removed it!